### PR TITLE
docs: Fix several PyLance issues

### PR DESCRIPTION
**Pull request by georgebisbas**

### CODE REVIEW

File: xdsl/ir.py [hunk 1]
```diff
@@ -38,7 +38,8 @@ def attributes(self) -> Iterator[type[Attribute]]:
 
     def __call__(self, ctx: MLContext) -> None:
         print(
-            "Calling a dialect in order to register it is deprecated and will soon be removed.",
+            "Calling a dialect in order to register it is deprecated"
```
Comment (math-fehr) [inline xdsl/ir.py:41]:
```suggestion
            "Calling a dialect in order to register it is deprecated "
```

File: xdsl/parser.py [hunk 2]
```diff
@@ -304,19 +306,23 @@ def backtracking(self, region_name: str | None = None):
         with some meta information in the history attribute.
 
         The backtracker accepts the following exceptions:
-         - ParseError: signifies that the region could not be parsed because of (unexpected) syntax errors
-         - AssertionError: this error should probably be phased out in favour of the two above
+         - ParseError: signifies that the region could not be parsed because of
+         (unexpected) syntax errors
```
Comment (math-fehr) [inline xdsl/parser.py:310]: Should we indent this line to make it more "readable" (not sure)

**Review by math-fehr — APPROVED** [review]
Nice! I just added two comments, but otherwise that's good.

**Review by superlopuh — APPROVED** [review]
Thank you! Was going to do a similar PR, great that you beat me to it :)

File: xdsl/parser.py (hunk 2 above)
Comment (georgebisbas) [reply xdsl/parser.py:310]: yes

**Review by webmiche — APPROVED** [review]
LGTM. Do you want to merge @georgebisbas? These PRs can have quite high conflict potential, so i personally prefer to merge them asap :)

**Pull request merged**